### DISCUSSION TITLE
Can't pass organisation-level secret to reusable workflow

### DISCUSSION THREAD

**Discussion by gnarea** [community/community#26735] [Actions]
<p>I’m trying to pass a GitHub Personal Access Token from a bot user to a <a href="https://docs.github.com/en/actions/learn-github-actions/reusing-workflows" rel="noopener nofollow ugc">reusable workflow</a>, but I can’t get it to work.</p>
<p><a href="https://github.com/relaycorp/relaynet-internet-gateway/blob/5e3a322a0499850b8097097a6a96f1c01f1b56bb/.github/workflows/prs.yml" rel="noopener nofollow ugc">Here’s my caller workflow</a>:</p>
<pre><code class="lang-yaml">name: Process PRs

on:
  pull_request:
    types:
      - opened

jobs:
  dependabot-automerge:
    uses: relaycorp/shared-workflows/.github/workflows/prs.yml@main
    secrets:
      githubToken: ${{ secrets.IRELAYBOT_PR_AUTOMATION_GITHUB_PAT }}
</code></pre>
<p><a href="https://github.com/relaycorp/shared-workflows/blob/83fb8b97ec734dae25567d088579b61bd57a9f95/.github/workflows/prs.yml" rel="noopener nofollow ugc">My called workflow</a>:</p>
<pre><code class="lang-yaml">name: Process pull requests
on:
  workflow_call:
    secrets:
      githubToken:
        required: true

jobs:
  dependabot:
    runs-on: ubuntu-latest
    if: ${{ github.actor == 'dependabot[bot]' &amp;&amp; github.event.action == 'opened' }}
    steps:
      - name: Report number of chars in GITHUB_TOKEN
        run: echo -n "${GITHUB_TOKEN}" | wc -c
        env:
          GITHUB_TOKEN: ${{ secrets.githubToken }}
      - name: Fetch metadata
        id: metadata
        uses: dependabot/fetch-metadata@v1.1.1
        with:
          github-token: ${{ secrets.githubToken }}
      - name: Trigger release if dependency is used in production
        if: ${{ steps.metadata.outputs.dependency-type == 'direct:production' }}
        run: gh pr edit "$PR_URL" --title "${OLD_TITLE/chore/fix}"
        env:
          OLD_TITLE: ${{ github.event.pull_request.title }}
          PR_URL: ${{ github.event.pull_request.html_url }}
          GITHUB_TOKEN: ${{ secrets.githubToken }}
      - name: Add automerge label
        run: gh pr edit "$PR_URL" --add-label "automerge"
        env:
          PR_URL: ${{ github.event.pull_request.html_url }}
          GITHUB_TOKEN: ${{ secrets.githubToken }}
</code></pre>
<p>But I get these errors when the action runs because the <code>githubToken</code> secret isn’t set:</p>
<pre><code class="lang-auto">Error: github-token is not set! Please add 'github-token: "${{ secrets.GITHUB_TOKEN }}"' to your workflow file
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/6/a653051fdb0366ebdd8249075932db8f3de6a453.png" data-download-href="/uploads/short-url/nJn82pVwbhEmRstcVLeWe8UP27N.png?dl=1" title="err" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/369863/181099645-35a2b161-f1fc-477f-b7d0-e0a34b71385e.png" alt="err" data-base62-sha1="nJn82pVwbhEmRstcVLeWe8UP27N" width="690" height="247" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/6/a653051fdb0366ebdd8249075932db8f3de6a453_2_690x247.png, https://aws1.discourse-cdn.com/github/original/3X/a/6/a653051fdb0366ebdd8249075932db8f3de6a453.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/6/a653051fdb0366ebdd8249075932db8f3de6a453.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/6/a653051fdb0366ebdd8249075932db8f3de6a453_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">err</span><span class="informations">889×319 27.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But I’m sure the secret exists and is readable by that repo:</p>
<p><img src="https://user-images.githubusercontent.com/369863/181099647-a7222c46-9c90-4255-b452-28aecb02d123.png" alt="secret" data-base62-sha1="ubtonfvuqJ08aTtqaERC2mby145" width="666" height="337"></p>
<p>What am I doing wrong? Or is this a bug?</p>
<p>Thanks.</p>

**Comment by gnarea** [comment]
<p>Does anyone know if this is a bug or I’m doing something wrong?</p>

**Comment by bboles** [comment]
<p>One issue I see is that you are trying to set <code>GITHUB_TOKEN</code> yourself.  You shouldn’t do this because it is automatically set at the beginning of each workflow:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/security-guides/automatic-token-authentication">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/security-guides/automatic-token-authentication" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/6370528/181099652-eae9e928-26da-461a-b077-b4a857ff4466.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/security-guides/automatic-token-authentication" target="_blank" rel="noopener nofollow ugc">Automatic token authentication - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
If you want to use your own token, you should use a different variable name.</p>

**Comment by gnarea** [comment]
<p>Sorry, I just got back to working on this issue and noticed I missed the notification for your reply.</p>
<p>I’m not setting <code>GITHUB_TOKEN</code> in the workflow though. I’m setting <code>githubToken</code>, which I then pass to the step as the env var <code>GITHUB_TOKEN</code>.</p>

**Comment by gnarea** [comment]
<p>The problem even persists if <a href="https://github.com/relaycorp/shared-workflows/commit/79903ff4e3297fb6e4cad00c9c282f753ee53267" rel="noopener nofollow ugc">I rename the variable</a>:</p>
<p><img src="https://user-images.githubusercontent.com/369863/181099654-6661080e-6a38-4c5a-8616-625f0f60f4d1.png" alt="token" data-base62-sha1="3CcBRtQ7tpufSCuj7fX9FYOLw8e" width="356" height="293"></p>
<p>So I don’t think this has something to do with <code>GITHUB_TOKEN</code> – It’s just that the org-level secret isn’t available. But why?</p>

**Comment by gnarea** [comment, marked as answer]
<p>I contacted GitHub support and they found the problem: The secret should’ve been set as a Dependabot secret at the org level… Not a regular secret.</p>
